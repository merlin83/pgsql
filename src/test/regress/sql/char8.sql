--  **************** testing built-in type char8 **************
--
-- all inputs are silently truncated at 8 characters
--

CREATE TABLE CHAR8_TBL(f1 char8);

INSERT INTO CHAR8_TBL(f1) VALUES ('ABCDEFGH');

INSERT INTO CHAR8_TBL(f1) VALUES ('abcdefgh');

INSERT INTO CHAR8_TBL(f1) VALUES ('ZYWZ410-');

INSERT INTO CHAR8_TBL(f1) VALUES ('343f%2a');

INSERT INTO CHAR8_TBL(f1) VALUES ('d34aas');

INSERT INTO CHAR8_TBL(f1) VALUES ('');

INSERT INTO CHAR8_TBL(f1) VALUES ('1234567890');


SELECT '' AS seven, CHAR8_TBL.*;

SELECT '' AS six, c.f1 FROM CHAR8_TBL c WHERE c.f1 <> 'ABCDEFGH';

SELECT '' AS one, c.f1 FROM CHAR8_TBL c WHERE c.f1 = 'ABCDEFGH';

SELECT '' AS three, c.f1 FROM CHAR8_TBL c WHERE c.f1 < 'ABCDEFGH';

SELECT '' AS four, c.f1 FROM CHAR8_TBL c WHERE c.f1 <= 'ABCDEFGH';

SELECT '' AS three, c.f1 FROM CHAR8_TBL c WHERE c.f1 > 'ABCDEFGH';

SELECT '' AS four, c.f1 FROM CHAR8_TBL c WHERE c.f1 >= 'ABCDEFGH';

SELECT '' AS seven, c.f1 FROM CHAR8_TBL c WHERE c.f1 ~ '.*';

SELECT '' AS zero, c.f1 FROM CHAR8_TBL c WHERE c.f1 !~ '.*';

SELECT '' AS four, c.f1 FROM CHAR8_TBL c WHERE c.f1 ~ '[0-9]';

SELECT '' AS three, c.f1 FROM CHAR8_TBL c WHERE c.f1 ~ '.*34.*';

DROP TABLE  CHAR8_TBL;
