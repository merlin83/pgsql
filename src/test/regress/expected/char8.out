QUERY: CREATE TABLE CHAR8_TBL(f1 char8);
QUERY: INSERT INTO CHAR8_TBL(f1) VALUES ('ABCDEFGH');
QUERY: INSERT INTO CHAR8_TBL(f1) VALUES ('abcdefgh');
QUERY: INSERT INTO CHAR8_TBL(f1) VALUES ('ZYWZ410-');
QUERY: INSERT INTO CHAR8_TBL(f1) VALUES ('343f%2a');
QUERY: INSERT INTO CHAR8_TBL(f1) VALUES ('d34aas');
QUERY: INSERT INTO CHAR8_TBL(f1) VALUES ('');
QUERY: INSERT INTO CHAR8_TBL(f1) VALUES ('1234567890');
QUERY: SELECT '' AS seven, CHAR8_TBL.*;
seven|f1      
-----+--------
     |ABCDEFGH
     |abcdefgh
     |ZYWZ410-
     |343f%2a 
     |d34aas  
     |        
     |12345678
(7 rows)

QUERY: SELECT '' AS six, c.f1 FROM CHAR8_TBL c WHERE c.f1 <> 'ABCDEFGH';
six|f1      
---+--------
   |abcdefgh
   |ZYWZ410-
   |343f%2a 
   |d34aas  
   |        
   |12345678
(6 rows)

QUERY: SELECT '' AS one, c.f1 FROM CHAR8_TBL c WHERE c.f1 = 'ABCDEFGH';
one|f1      
---+--------
   |ABCDEFGH
(1 row)

QUERY: SELECT '' AS three, c.f1 FROM CHAR8_TBL c WHERE c.f1 < 'ABCDEFGH';
three|f1      
-----+--------
     |343f%2a 
     |        
     |12345678
(3 rows)

QUERY: SELECT '' AS four, c.f1 FROM CHAR8_TBL c WHERE c.f1 <= 'ABCDEFGH';
four|f1      
----+--------
    |ABCDEFGH
    |343f%2a 
    |        
    |12345678
(4 rows)

QUERY: SELECT '' AS three, c.f1 FROM CHAR8_TBL c WHERE c.f1 > 'ABCDEFGH';
three|f1      
-----+--------
     |abcdefgh
     |ZYWZ410-
     |d34aas  
(3 rows)

QUERY: SELECT '' AS four, c.f1 FROM CHAR8_TBL c WHERE c.f1 >= 'ABCDEFGH';
four|f1      
----+--------
    |ABCDEFGH
    |abcdefgh
    |ZYWZ410-
    |d34aas  
(4 rows)

QUERY: SELECT '' AS seven, c.f1 FROM CHAR8_TBL c WHERE c.f1 ~ '.*';
seven|f1      
-----+--------
     |ABCDEFGH
     |abcdefgh
     |ZYWZ410-
     |343f%2a 
     |d34aas  
     |        
     |12345678
(7 rows)

QUERY: SELECT '' AS zero, c.f1 FROM CHAR8_TBL c WHERE c.f1 !~ '.*';
zero|f1
----+--
(0 rows)

QUERY: SELECT '' AS four, c.f1 FROM CHAR8_TBL c WHERE c.f1 ~ '[0-9]';
four|f1      
----+--------
    |ZYWZ410-
    |343f%2a 
    |d34aas  
    |12345678
(4 rows)

QUERY: SELECT '' AS three, c.f1 FROM CHAR8_TBL c WHERE c.f1 ~ '.*34.*';
three|f1      
-----+--------
     |343f%2a 
     |d34aas  
     |12345678
(3 rows)

